src/hooks/useDeviceSize.js

```diff
@@ -0,0 +1,26 @@
+import { useEffect, useState } from 'react';
+
+const useMediaQuery = (query) => {
+  const [matches, setMatches] = useState(false);
```

Comment:
`matches` 초기값을 `false`로 설정하고 있는데, `window.matchMedia(query).matches`에서 바로 초기값을 가져오는 것이 좋습니다. 그렇지 않으면 첫 렌더링 시 false가 먼저 반환되고, `useEffect`가 실행된 후에 값을 업데이트하게 되어, 초기 렌더링 시에 불필요한 re-rendering이 발생할 수 있습니다.

다음과 같이 초기값을 설정하면 첫 렌더링부터 정확한 값으로 `matches`가 설정되어 불필요한 렌더링을 피할 수 있습니다.
```suggestion
  const [matches, setMatches] = useState(() => window.matchMedia(query).matches);
```